src/base/misc/fatfs.c

```diff
@@ -421,7 +421,9 @@ int read_boot(fatfs_t *f, unsigned char *b)
     if(bpb->v340.signature == 0x28 || bpb->v340.signature == 0x29) {
       bpb->v340.drive_number = f->drive_num;
     } else {
-      b[0x1fd] = f->drive_num;
+      if (b[0x1fd] >= 0x80 && b[0x1fd] <= 0x83) { // Only those that have it
+        b[0x1fd] = f->drive_num;
```

Comment:
I'll explain. In the case where we are reusing a boot.blk provided in a file by the user and it is a version that doesn't have an easily identifiable BPB version i.e. less than v 3.40, we would like to write the drive number into location 0x01fd which is used by some, but not all early versions of DOS. In order to be safe, we test the old boot block to see if that location has valid hard drive number(0x80.. 0x83) already there in that location. if so, we update it with the current drive number. We can't just write it every time without checking, since some boot blocks have other data in that area. On one occasion a boot block had a value of 0x00 in that position which turned out to be the terminator of a string. For that reason we need to be quite narrow about the range of values we could consider to be a valid drive. Have you a better idea how I could accomplish this, or is adding an explanatory comment okay?